packages/amplify-graphql-schema-generator/src/__tests__/schema-overrides.test.ts

```diff
@@ -397,6 +398,229 @@ describe('apply schema overrides for models with refersTo', () => {
     const updatedDocument = applySchemaOverrides(document, editedDocument);
     stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
   });
+
+  it('should not allow duplicate model name mappings in edited schema', () => {
```

Comment:
Can we add a test case when refersTo is used on a model which is used as output type on a custom query using @sql directive?

Reply:
To pile onto this: we should have a positive case (refersTo maps to an output type that doesn't conflict) as well as failure case (refersTo maps to an output type that conflicts)